### DISCUSSION TITLE
Update an existing User record by connecting it to a variable amount of existing Post records

### DISCUSSION THREAD

**Discussion by LoreboxGGG** [prisma/prisma#3961] [Q&A] [answered]
Hi,
The [Prisma Documentation](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/relation-queries#update-an-existing-user-record-by-connecting-it-to-two-existing-post-records) covers how to Update an existing User record by connecting it to two existing Post records:

```
const user = await prisma.user.update({
  where: { email: 'alice@prisma.io' },
  data: {
    posts: {
      connect: [{ id: 24 }, { id: 42 }],
    },
  },
})
```

However, I am looking for a way to connect an unknown amount of existing Post records. 
For example, if an array of post ids is passed in as an argument, I want to connect all of those Posts to the User.

Is this possible to do? I've tried various syntax, but get an invalid value.

**Comment by LoreboxGGG** [comment, marked as answer]
Adding 
```
input PostInput {
  id: Int!
}
``` 

to the schema.graphql and passing that as an argument in the mutation allowed me to do what I wanted.

```
const user = await prisma.user.update({
  where: { email: 'alice@prisma.io' },
  data: {
    posts: {
      connect: args.posts,
    },
  },
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
